distributor.go

```diff
@@ -189,30 +193,45 @@ func metricNameFromLabelMatchers(matchers ...*metric.LabelMatcher) (model.LabelV
 func (d *Distributor) Query(ctx context.Context, from, to model.Time, matchers ...*metric.LabelMatcher) (model.Matrix, error) {
 	var result model.Matrix
 	err := instrument.TimeRequestHistogram("duration", d.queryDuration, func() error {
-		metricName, err := metricNameFromLabelMatchers(matchers...)
-		if err != nil {
-			return err
-		}
+		fpToSampleStream := map[model.Fingerprint]*model.SampleStream{}
+
+		// Fetch samples from all ingesters and group them by fingerprint (unsorted
+		// and with overlap).
```

Comment:
We can, but it's more complexity - it's probably not going to be really noticeable at current ingester scale. Reevaluate later?